### ISSUE TITLE
Handling of true/false in Style

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#815]
## Summary

Some shape attributes, such as `Line:startArrowhead` take a boolean value `true|false`.  This issue proposes that either (i) numerical values be interpretable as boolean values, (ii) add an if-then-else construct to Style that is capable of returning a boolean value, or (iii) both.

## Problem

Currently, there is no mechanism for setting these boolean values based on the result of a floating point calculation.  Consider, for instance, [these lines](https://github.com/penrose/penrose/blob/main/examples/exterior-algebra/exterior-algebra.sty#L105) of the `exterior-algebra` example, which draw a clockwise or counter-clockwise arrow:

```
-- draw an orientation marker
   scalar w.c = (u.p+v.p)/2. -- center
   scalar R = .75 * minWidth/2. -- radius
   vec2 x0 = w.c + R*unit(u.p-v.p) -- arc start
   vec2 x1 = w.c + R*unit(v.p-u.p) -- arc end
   scalar cw = .5*sign( cross2D(u.p,v.p) ) + .5 -- clockwise (1) or not (0)
   shape w.marker = Path {
      d: arc( "open", x0, x1, (R,R), 0., 0, cw )
      fillColor: none()
      strokeColor: w.solidColor
      strokeWidth: .75*Global.lineThickness
      startArrowhead: true
      arrowheadSize: .5
   }
```

The scalar `cw` ultimately holds a value of 0 or 1.  But setting `startArrowhead: cw` does not correctly flag the arrowhead as on or off.  There is also no way, currently, to use a conditional (if-then-else) statement in Style.

In this case, fortunately, there is another option: set a flag of the SVG `<path>` tag to 1, to achieve the same effect.  But in general such a solution is not always available.

## Proposed solutions

**Interpret numbers as booleans.** One possibility is to allow numerical values to be interpreted as boolean values, with `0` (or floating point `0.0`) evaluating to `false` and all other values evaluating to true.  Basically we would just follow the rules of the ambient language (TypeScript/JavaScript).

**Provide if-then-else.** Another possibility is to enrich the language with a feature that we want anyway: conditional assignment.  Note that conditional _assignment_ is different from conditional _execution_: we do not want to be able to invoke arbitrary blocks of Style code based on a condition.  Rather (for now) we only need to be able to make assignments of the form

```
scalar x = c > 0 ? a; b
```

In other words, assign a value based on whether a given condition holds.  For conditions and return values purely involving floating-point numbers, this mechanism is already implementable via the `ifCond` function in `Autodiff.ts` (which also provides derivatives).  But at present there is no way to write a conditional that returns a boolean value, e.g.,

```
startArrowhead: c>0? true; false
```

**Comment by samestep** [comment]
> **Interpret numbers as booleans.**

I veto this :)
